### PR TITLE
[IT-937] Add cloudformation template for HTAN bucket

### PR DESCRIPTION
**Pull request by zaro0508**
Add a cloudformation template to automate deployment of
HTAN buckets that will work with the s3-synapse-sync lambda[1]

[1] https://github.com/Sage-Bionetworks/s3-synapse-sync

Depends on Sage-Bionetworks/s3-synapse-sync#3

### CODE REVIEW

**Comment by zaro0508** [comment]
please review @clarisse-lau

File: templates/htan-synapse-bucket.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    Properties:
+      Bucket: !Ref S3Bucket
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          -
+            Sid: "ReadAccess"
+            Effect: "Allow"
+            Principal:
+              AWS: !Ref S3UserARNs
```
Comment (clarisse-lau) [inline templates/htan-synapse-bucket.yaml:90]: Should the Synapse root arn be included in this principal(and that of the "WriteAccess" statement)?

File: templates/htan-synapse-bucket.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            Action:
+              - "s3:ListBucket*"
+              - "s3:GetBucketLocation"
+            Resource: !Split
+              - ","
+              - !Sub
+                - "${list},${SynapseRootArn}"
+                - list: !Join [",",!Ref "S3UserARNs"]
```
Comment (clarisse-lau) [inline templates/htan-synapse-bucket.yaml:98]: wondering if the resource for “ReadAccess” and “WriteAccess” statements should be the S3 bucket?

File: templates/htan-synapse-bucket.yaml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+          -
+            Sid: "LambdaAccess"
+            Effect: Allow
+            Principal:
+              AWS: !Ref S3SynapseSyncFunctionRoleArn
```
Comment (clarisse-lau) [inline templates/htan-synapse-bucket.yaml:116]: I believe S3UserARNs will also need these permissions

Comment (zaro0508) [reply templates/htan-synapse-bucket.yaml:116]: "WriteAccess" already provides `s3:*Object*` to S3UserARNs so I don't think we need it here.

File: templates/htan-synapse-bucket.yaml (hunk 1 above)
Comment (zaro0508) [reply templates/htan-synapse-bucket.yaml]: done

File: templates/htan-synapse-bucket.yaml (hunk 2 above)
Comment (zaro0508) [reply templates/htan-synapse-bucket.yaml]: done

File: templates/htan-synapse-bucket.yaml (hunk 3 above)
Comment (clarisse-lau) [reply templates/htan-synapse-bucket.yaml:116]: Right. Looks good to me then

**Pull request merged**